scex-util/src/main/scala/com/avsystem/scex/util/EnrichedDate.scala

```diff
@@ -49,27 +50,27 @@ final class EnrichedDate(private val wrapped: Date) extends AnyVal {
   def millis: Long = wrapped.getTime
 
   @Documentation("Returns the seconds field of the date expressed in milliseconds.")
-  def millisOfSecond: Int = dateTime.getMillisOfSecond
+  def millisOfSecond: Int = instant.get(ChronoField.MILLI_OF_SECOND)
   @Documentation("Returns the seconds field of the date.")
-  def secondOfMinute: Int = dateTime.getSecondOfMinute
+  def secondOfMinute: Int = instant.get(ChronoField.SECOND_OF_MINUTE)
   @Documentation("Returns the number of seconds which passed since 00:00 for the date.")
-  def secondOfDay: Int = dateTime.getSecondOfDay
+  def secondOfDay: Int = instant.get(ChronoField.SECOND_OF_DAY)
   @Documentation("Returns the minutes field of the date.")
-  def minuteOfHour: Int = dateTime.getMinuteOfHour
+  def minuteOfHour: Int = instant.get(ChronoField.MINUTE_OF_HOUR)
   @Documentation("Returns the number of minutes which passed since 00:00 for the date.")
-  def minuteOfDay: Int = dateTime.getMinuteOfDay
+  def minuteOfDay: Int = instant.get(ChronoField.MINUTE_OF_DAY)
   @Documentation("Returns the number of hours which passed since 00:00 for the date.")
-  def hourOfDay: Int = dateTime.getHourOfDay
+  def hourOfDay: Int = instant.get(ChronoField.HOUR_OF_DAY)
   @Documentation("Returns the day field of the date.")
-  def dayOfMonth: Int = dateTime.getDayOfMonth
+  def dayOfMonth: Int = instant.get(ChronoField.DAY_OF_MONTH)
 
   @Documentation("Returns a numeric value for day of the week of the date, where 1 - Monday, 7 - Sunday.")
-  def dayOfWeek: Int = dateTime.getDayOfWeek
+  def dayOfWeek: Int = instant.get(ChronoField.DAY_OF_WEEK)
 
   @Documentation("Returns a numeric value for the day of the year.")
-  def dayOfYear: Int = dateTime.getDayOfYear
+  def dayOfYear: Int = instant.get(ChronoField.DAY_OF_YEAR)
 
   @Documentation("Returns a numeric value for the month of the year, where 1 - January, 12 - December.")
-  def monthOfYear: Int = dateTime.getMonthOfYear
+  def monthOfYear: Int = instant.get(ChronoField.MONTH_OF_YEAR)
```

Comment:
Unfortunately:
```
Unsupported field: MonthOfYear
java.time.temporal.UnsupportedTemporalTypeException: Unsupported field: MonthOfYear
	at java.base/java.time.Instant.get(Instant.java:567)
	at com.avsystem.scex.util.EnrichedDate$.monthOfYear$extension(EnrichedDate.scala:74)
	at com.avsystem.scex.util.EnrichedDateTest.$anonfun$new$7(EnrichedDateTest.scala:44)
```

Reply:
Consider adding tests for these methods

Reply:
Done. Note that the tests depend on the timezone, so I fixed it to Europe/Warsaw.